executor/builder.go

```diff
@@ -998,6 +998,14 @@ func (b *executorBuilder) buildHashJoin(v *plannercore.PhysicalHashJoin) Executo
 		joinType:      v.JoinType,
 		isOuterJoin:   v.JoinType.IsOuterJoin(),
 		innerEstCount: v.Children()[v.InnerChildIdx].StatsCount(),
+		outerHashJoin: v.OuterHashJoin,
+	}
+	// reverse the inner and the outer
+	if e.outerHashJoin {
```

Comment:
Why do we need to reverse the outer and inner side?
We can use the outer child to build the hash table directly if outerHashJoin is true without any modification on the metadata of HashJoinExec.

Reply:
> Why do we need to reverse the outer and inner side?
> We can use the outer child to build the hash table directly if outerHashJoin is true without any modification on the metadata of HashJoinExec.

If following your way, the build and the probe have to change a lot of codes. Here, I just reverse the outer and the inner within the join, the output of the join remains the same. The previous codes can be reused as many as possible.